arangod/GeneralServer/AsyncJobManager.cpp

```diff
@@ -193,7 +192,8 @@ Result AsyncJobManager::cancelJob(AsyncJobResult::IdType jobId) {
   }
 
   bool ok = true;
-  RestHandler* handler = it->second.second._handler;
+  std::shared_ptr<RestHandler> handler = it->second.second._handler;
```

Comment:
Use a reference here to avoid incrementing/decrementing the ref count (which are expensive atomic operations).